redbot/cogs/downloader/downloader.py

```diff
@@ -499,24 +499,21 @@ def __init__(self, bot: Red):
         """Update all repos, or ones of your choosing."""
         async with ctx.typing():
             updated: Set[str]
-            if not repos:
-                updated = {repo.name for repo in await self._repo_manager.update_all_repos()}
-            else:
-                updated = set()
-                for repo in repos:
-                    old, new = await repo.update()
-                    if old != new:
-                        updated.add(repo.name)
+
+            updated_repos, failed = await self._repo_manager.update_repos(repos)
+            updated = {repo.name for repo in updated_repos}
 
             if updated:
                 message = _("Repo update completed successfully.")
                 message += _("\nUpdated: ") + humanize_list(tuple(map(inline, updated)))
             elif repos is None:
```

Comment:
This `if` is inaccessible, it isn't like that because of changes you did, but I'm mentioning it since you did fix it in one of previous versions of this PR.
```suggestion
             elif not repos:
```

Reply:
https://github.com/Cog-Creators/Red-DiscordBot/pull/3080#discussion_r349374222

Reply:
The comment you linked doesn't seem to be related to the issue I'm talking about here. If I'm correct, Flame was talking just about the string that is sent in `else` block and I'm talking about inaccessible code in the `elif` above.

Reply:
Before changes of this PR it looked like this:
`elif repos is None:`
Thru out the PR I have changed it to
`elif not repos:`
Flame commented that the `repos` can contain multiple repos or one (`ctx, *, repos`) I didn't notice that and changed it back to:
`elif repos is None`

Why should it be inaccessible ?